### PR TITLE
fix #166016 slur on grace after wrong

### PR DESCRIPTION
**Pull request by lyrischesich**
The algorithm for finding the first and last note of the selection ends up with the two notes reversed. This happens only if a gracenote after is part of the selection. So I decided to test for this case and simply commute the two values.

### CODE REVIEW

File: mscore/scoreview.cpp [hunk 1]
```diff
@@ -4487,6 +4487,11 @@ void ScoreView::cmdAddSlur()
                   return;
             if (firstNote == lastNote)
                   lastNote = 0;
+            else if (firstNote->chord()->isGraceAfter()) {
```
Comment (MarcSabatella) [inline mscore/scoreview.cpp]: Do we understand *why* the notes are reversed, and are we sure it is both a) always true for grace notes after, and b) never *not* true for grace notes after?  Personally, I'd rather see the the basic algorithm fixed rather than patched at the end.

Comment (lyrischesich) [reply mscore/scoreview.cpp]: I agree with your doubts and revised the algorithm. Now it should cover every possible case, including grace notes before and after.

**Comment by Jojo-Schmitz** [comment]
Please make sure the issue numbers are mentioned in the commit title. having them only in the PR title don't result in the issues to automagically get marked as fixed once the PR got merged.

**Comment by lyrischesich** [comment]
I changed the commit titles

**Comment by lasconic** [comment]
I guess we don't really need the two commits here right ? Can they be squashed?
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by lyrischesich** [comment]
Shure.

**Comment by lasconic** [comment]
In general we don't accept PR on the 2.2 branch. The development happens on the master branch only and fixes  that apply to 2.2 are manually cherry picked to the 2.2 branch. If you plan to do other PR, please use the master branch.

**Comment by Jojo-Schmitz** [comment]
Unless the PR applies to 2.2 only and not to master at all, or needs to be very different between the 2

**Comment by lyrischesich** [comment]
So I should close this and do another PR on master? Since the original code looks very similar in 2.2. an master.

**Comment by Jojo-Schmitz** [comment]
I believe @lasconic would let this one here pass as is (once squashed) ;-), as far as I understood he's talking about future PRs from you

**Comment by Jojo-Schmitz** [comment]
Outch that went horribly wrong. Don't merge, but rebase

**Comment by Jojo-Schmitz** [comment]
BTW: this makes me wonder whether #3381 was really meant for 2.2 only?

**Comment by Jojo-Schmitz** [comment]
Oh, OK, and for that there is no fix yet, right? I'll reopen that issue then

**Comment by lasconic** [comment]
If find the logic in this PR hard to follow. Here is my take on the same issue: https://github.com/musescore/MuseScore/pull/3387 What do you think?
